bind/motor-mobile/bind.go

```diff
@@ -70,96 +76,110 @@ func Login(buf []byte) ([]byte, error) {
 
 	// decode request
 	var request mt.LoginRequest
-	if err := json.Unmarshal(buf, &request); err != nil {
+	if err := request.Unmarshal(buf); err != nil {
 		return nil, fmt.Errorf("error unmarshalling request: %s", err)
 	}
 
 	if res, err := instance.Login(request); err == nil {
-		return json.Marshal(res)
+		return res.Marshal()
 	} else {
 		return nil, err
 	}
 }
 
+func Connect() error {
+	if instance == nil {
+		return errWalletNotExists
+	}
+	return instance.Connect()
+}
+
 func CreateSchema(buf []byte) ([]byte, error) {
 	if instance == nil {
 		return nil, errWalletNotExists
 	}
 
 	var request mt.CreateSchemaRequest
-	if err := json.Unmarshal(buf, &request); err != nil {
+	if err := request.Unmarshal(buf); err != nil {
 		return nil, fmt.Errorf("unmarshal request: %s", err)
 	}
 
 	if res, err := instance.CreateSchema(request); err == nil {
-		return json.Marshal(res)
+		return res.Marshal()
 	} else {
 		return nil, err
 	}
 }
 
-func QueryWhatIs(buf []byte) ([]byte, error) {
+// Query is a generic query function that can be used to query any object, bucket, or DIDDocument in the Sonr network.
+func Query(buf []byte) ([]byte, error) {
 	if instance == nil {
 		return nil, errWalletNotExists
 	}
 
-	var request mt.QueryWhatIsRequest
-	if err := json.Unmarshal(buf, &request); err != nil {
+	var request mt.QueryRequest
+	if err := request.Unmarshal(buf); err != nil {
 		return nil, fmt.Errorf("unmarshal request: %s", err)
 	}
 
-	if res, err := instance.QueryWhatIs(context.Background(), request); err == nil {
-		return json.Marshal(res)
+	if res, err := instance.Query(request); err == nil {
+		return res.Marshal()
 	} else {
 		return nil, err
 	}
 }
 
-// Address returns the address of the wallet.
-func Address() string {
+// IssuePayment creates a send/receive token request to the specified address.
+func IssuePayment(buf []byte) ([]byte, error) {
 	if instance == nil {
-		return ""
+		return nil, errWalletNotExists
+	}
+
+	var request mt.PaymentRequest
+	if err := request.Unmarshal(buf); err != nil {
+		return nil, fmt.Errorf("unmarshal request: %s", err)
+	}
+
+	if res, err := instance.SendTokens(request); err == nil {
+		return res.Marshal()
+	} else {
+		return nil, err
 	}
+}
+
+// Stat returns general information about the Motor node its wallet and accompanying Account.
+func Stat() ([]byte, error) {
```

Comment:
I think it's unlikely that address, balance and did document will be needed at the same time. Having individual functions for them makes more sense imo.

Reply:
Its been way easier to handle the data on the client side by utilizing a protobuf for this rather than returning raw strings - the added benefit of extension methods for these types like I am with `motor-flutter` gives further customizability. I think we should vote on this.